src/manage_nsfs/nc_lifecycle.js

```diff
@@ -3,20 +3,38 @@
 
 const dbg = require('../util/debug_module')(__filename);
 const _ = require('lodash');
+const path = require('path');
 const util = require('util');
 const P = require('../util/promise');
 const config = require('../../config');
 const nb_native = require('../util/nb_native');
 const NsfsObjectSDK = require('../sdk/nsfs_object_sdk');
+const native_fs_utils = require('../util/native_fs_utils');
 const ManageCLIError = require('./manage_nsfs_cli_errors').ManageCLIError;
 const { throw_cli_error, get_service_status, NOOBAA_SERVICE_NAME } = require('./manage_nsfs_cli_utils');
 
-// TODO: 
+// TODO:
 // implement 
 // 1. notifications
 // 2. POSIX scanning and filtering per rule
 // 3. GPFS ILM policy and apply for scanning and filtering optimization
 
+const CLUSTER_LOCK = 'cluster.lock';
+
+/**
+ * run_lifecycle_under_lock runs the lifecycle workflow under a file system lock
+ * lifecycle workflow is being locked to prevent multiple instances from running the lifecycle workflow
+ * @param {import('../sdk/config_fs').ConfigFS} config_fs 
+ */
+async function run_lifecycle_under_lock(config_fs) {
+    const lock_path = path.join(config_fs.config_root, CLUSTER_LOCK);
```

Comment:
Will configfs always be on a distrubuted FS like GPFS, Ceph, etc? This is true for DBS3 but I am unaware of other deployments.

Reply:
you are right, I did this change on https://github.com/noobaa/noobaa-core/pull/8860/files and now took it to this PR as well

Reply:
This change was a mistake, we must have the lock and timestamp files in the config directory which is a shared root between all the nodes